src/box/lua/upgrade.lua

```diff
@@ -1236,8 +1236,39 @@ local function revoke_write_access_on__collation_from_role_public()
     _priv:delete{PUBLIC, 'space', box.schema.COLLATION_ID}
 end
 
+local function convert_sql_constraints_to_tuple_constraints()
+    local _fk = box.space[box.schema.FK_CONSTRAINT_ID]
+    local _ck = box.space[box.schema.CK_CONSTRAINT_ID]
+    log.info("convert constraints from _ck_constraint and _fk_constraint")
+    for _, v in _fk:pairs() do
+        local space = box.space[v.child_id]
+        local mapping = setmap({})
+        for i, id in pairs(v.child_cols) do
+            mapping[id + 1] = v.parent_cols[i] + 1
+        end
+        local fk = space.foreign_key or {};
```

Comment:
AFAIK in the upgrade script the internal alter triggers are disabled. Which means the Lua space objects are not updated on any DDL. And that means `space.foreign_key` remains unchanged this entire loop. Same for `space.constraint`. Or how does it work?